File: services/agent/src/routes/docker.js
const express = require("express");
const Docker = require("dockerode");

const router = express.Router();

const docker = new Docker({
    socketPath: "/var/run/docker.sock",
});

router.get("/", async (req, res) => {
    try {
        const containers = await docker.listContainers({
            all: true,
        });

        const result = containers.map((container) => ({
            id: container.Id,
            name: container.Names[0]?.replace("/", ""),
            image: container.Image,
            state: container.State,
            status: container.Status,
        }));

        res.json(result);
    } catch (error) {
        console.error(error);

        res.status(500).json({
            error: "Failed to fetch docker containers",
        });
    }
});

module.exports = router;

File: services/agent/app/routes/docker.py
import docker
from fastapi import APIRouter

router = APIRouter()

client = docker.from_env()


@router.get("/docker")
def get_containers():
    containers = []

    for container in client.containers.list(all=True):
        containers.append({
            "id": container.short_id,
            "name": container.name,
            "image": container.image.tags,
            "status": container.status
        })

    return containers